docs/blog/2024-05-07-nx-19-release.md

```diff
@@ -0,0 +1,201 @@
+
+To see how we support this new feature, checkout the json view of our project again:
+
+```shell
+nx show project e2e-angular --json | jq
+```
+
+![Highlighting target metadata](/blog/images/2024-05-07/highlight-target-metadata.webp)
+
+Notice how tasks now have a `metadata` property including `technologies` and `description` properties. As with the Atomizer, any custom tools can also take advantage of these properties to enable this behavior in Nx and Nx Cloud tooling!
+
+## Generators to Migrate to Project Crystal
+
+As part of Nx 19, we've included migrations for our Playwright, Cypress, and ESLint plugins to allow you to easily opt into Project Crystal features!
+
+To migrate your workspace, you can use the command:
+
+```shell
+nx generate convert-to-inferred
+```
+
+Before running this command you could expect your project detail view to look like this:
+
+```shell
+nx show project my-react-app-e2e --web
+```
+
+![Before Migration](/blog/images/2024-05-07/before-migration.webp)
+
+And after running the generator, you can expect your project detail view to look like this:
+
+![After Migration](/blog/images/2024-05-07/after-migration.webp)
+
+Notice that in addition to reducing the configuration in your `project.json` file, this generator will also turn on the Nx Atomizer for your Cypress and Playwright projects, so Nx will always shard your end-to-end tasks based on the tests present in your workspace going forward.
+
+As a note, if you cannot see `e2e-ci` tasks after migrating your cypress projects, you can [take a closer look here](https://nx.dev/nx-api/cypress/documents/overview#splitting-e2e-tasks-by-file) at how to configure them correctly.
+
+## BREAKING CHANGE: Updating Bundled Environment Variables: `NX_` to `NX_PUBLIC_`
+
+An important update for folks using environment variables in their builds!
+
+Previously, when building projects your projects in an Nx workspace, our builds were setup to include any environment variables prefixed with `NX_` and include them in the built artifact so they would be accessible there. This is often used for setting things like urls for different versions of external services to hit - for example, sending requests to your local development server while developing, but pointing to the production service for your production build.
+
+Going forward, these environment variables will need to be prefixed with `NX_PUBLIC_`. This is a security consideration - by requiring a more specific prefix we're making sure that no secrets end up in your production builds unintentionally!
+
+Because of the nature of this change we will NOT be providing an automatic migration. So be sure to manually update any environment variables that you want bundled into your builds from `NX_` to `NX_PUBLIC_`.
+
+## General Crystal Polishing
+
+Along with the features above (many of which further build upon Nx Project Crystal), we've done quite a bit since Nx 18 to polish the features that we introduced with Project Crystal.
```

Comment:
During the time since Nx 18 was released, we also started dogfooding Project Crystal in the Nx, Nx Console, and our closed-source repo. In doing so, we've uncovered and fixed many issues.

Over the the last few months, we've noticed an influx of Github Issues which we previously did not allocate enough time to handling. To adjust for the influx of issues, the team is going to be doubling their time focusing on maintenance.

In addition to spending more time, we are also adding more process to make our triage more efficient. Github Issues will now be assigned directly to members of the Core Team which allows every member to spend less time figuring out what to fix, and more time actually fixing issues.